joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/utils/AstGenRunner.scala

```diff
@@ -23,7 +23,17 @@ object AstGenRunner {
 
   private val LineLengthThreshold: Int = 10000
 
-  private val NODE_OPTIONS: Map[String, String] = Map("NODE_OPTIONS" -> "--max-old-space-size=8192")
+  private def getMaxHeapSizeInMB: Long = {
+    val maxHeapSizeInBytes = Runtime.getRuntime.maxMemory()
+    val maxHeapSizeInMB    = maxHeapSizeInBytes / (1024 * 1024)
+    maxHeapSizeInMB // nodejs expects MB
+  }
+
+  private def nodejsSpaceSize: Int = {
+    Math.max((0.75 * getMaxHeapSizeInMB).toInt, 8192) // as default we set 8G minimum
```

Comment:
I understand that this isn't really a change, but I'm still wondering if setting a (rather high) minimum is the right idea, given that it may exceed the free memory, and/or even the physically available memory.

Reply:
I would assume this memory will be long freed when it needs to be reclaimed by java again. This is done under the assumption the end user will set up java mem thinking that this is all it takes for the cpg process